service/history/queue/timer_queue_processor.go

```diff
@@ -425,11 +432,14 @@ func (t *timerQueueProcessor) completeTimerLoop() {
 					return
 				}
 
+				// Reset the retryTimer for the delay between attempts
+				retryDuration := time.Duration(attempt*100) * time.Millisecond
+				retryTimer.Reset(retryDuration)
```

Comment:
this was true before too, but I think this isn't actually intended / may be worth branching on to be explicit:
on the first loop this will be `Reset(0)` because `attempt == 0`, so it retries immediately.  (and allows shutdown at least 50% of the time, which is not bad / may be worth it for that alone)

I'm not sure if that's actually desirable or not.  should this be `(attempt+1)*100`?

Reply:
it's hard to tell the original intent. Immediate retry might be intentional but I doubt that. 
@Shaddoll Let's leave a note in the code about this initial backoff 0ms situation to remind ourselves while revisiting history queues